### PR TITLE
 Doc rework (user guide, part 6)

### PR DESCRIPTION
**Pull request by stephenfin**
Add setuptools, i18n, websupport and theming documents to the user guide. As before, this is mostly done by shuffling around existing content.

### CODE REVIEW

**Comment by ArtFlag** [comment]
Cool PR, it's nice to improve the docs! Also I have a quick question about the docs in general:

I have seen that you plan on editing the docs in of the road maps topics. I also want to add more extension examples/tutorials soon and in order to avoid conflicting changes I'd like to ask if you are planning on updating the extension section before the end of the year?

**Comment by stephenfin** [comment]
@ArtFlag I plan to add a developer guide which will include a reworked extension guide, but it could be a while before I get to this. Feel free to make your changes and I'll revise mine accordingly

File: doc/usage/advanced/intl.rst [hunk 1]
```diff
@@ -63,125 +63,133 @@ be translated you need to follow these instructions:
...
    .. code-block:: console
 
       $ sphinx-intl update -p _build/gettext -l de -l ja
 
-   Done. You got these directories that contain po files:
+   Once completed, the generated po files will be placed in the below
+   directories:
+
+   * ``./locale/de/LC_MESSAGES/``
+   * ``./locale/ja/LC_MESSAGES/``
```
Comment (tk0miya) [inline doc/usage/advanced/intl.rst:117]: On my local, sphinx-intl generates po files under `locales` (plural) directory. I think this should be updated.

**Review by tk0miya — APPROVED** [review]
LGTM!

**Pull request merged**